CMakeLists.txt

```diff
@@ -158,6 +158,11 @@ if (FMT_PEDANTIC)
   target_compile_options(fmt PRIVATE ${PEDANTIC_COMPILE_FLAGS})
 endif ()
 
+if (NOT ${CMAKE_VERSION} VERSION_LESS 3.8)
+  target_compile_features(fmt INTERFACE cxx_std_11)
```

Comment:
This is a bit too strict. {fmt} requires only a small subset of C++11 features, mostly variadic templates, traits and auto.

Reply:
Sadly there is no compile feature for `<type_traits>`, however there are `cxx_auto_type` and `cxx_variadic_templates`, which are even available since CMake 3.1.
So if we assume that any compiler with those 2 featutres has type_traits, I think we could remove the CMake version check, and just use those 2 compile features ?

Reply:
Thinking more of it, what is the benefit of `target_compile_features`? If it only adds configuration-time checks, then I don't think it is needed because compilation will fail one way or another if some compiler features are not available. So adding this will only add more work at configuration time and potential for mismatch between the source and configuration.

Reply:
It adds the -std=c++11 flag.

Reply:
Ah, makes sense. What if the user wants to specify higher standard version though?

Reply:
I'm pretty sure it just works, that's the whole idea behind it (as opposed to manually setting the cxx standard variable or similar)

Reply:
> we could remove the CMake version check, and just use those 2 compile features ?

Sounds good then.